main.py

```diff
@@ -229,7 +228,7 @@ def escape_path(path):
 		short_anim_arg = ',short_animation' if settings.settings['short_anim'] else ''
 		hdd_lock_arg = ',locked' if settings.settings['hdd_locked'] else ''
 		sys_memory = settings.settings['sys_memory'].split(' ')[0]+'M'
-		accelerator_arg = Xqemu.generateAcceleratorArg(settings.settings['use_accelerator'])
+		irqchip_arg = ',kernel_irqchip=off' if sys.platform == 'linux2' else ''
```

Comment:
[The documentation for `sys.platform` says](https://docs.python.org/3/library/sys.html#sys.platform): "at the time when Python was built."
So this sounds to me, as if cross-platform builds would return bad information - might not be an issue though?

However, it also says - and this is critical:

> Changed in version 3.3: On Linux, sys.platform doesn’t contain the major version anymore. It is always 'linux', instead of 'linux2' or 'linux3'. Since older Python versions include the version number, it is recommended to always use the startswith idiom presented above.

---

I also considered going back to the `platform` module - but that seems to have been around since Python 1.x with most documentation being for Python 2.x. I believe `sys` is indeed the module we should use to get information from. It provides very similar features.

Reply:
you are right it wouldnt change after a build. so the only real way of determining platform then would be the platform module.

Reply:
I'd still stick to `sys.platform` (see last paragraph).

We should only verify what happens in mingw etc, or maybe look for alternatives in `sys` (as it also supports runtime checks to get the Windows version etc., so maybe it has another function similar to `sys.platform`).

Reply:
need to tinker with it a bit to see if we can get system info after building